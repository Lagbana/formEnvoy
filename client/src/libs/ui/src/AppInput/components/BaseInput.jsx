/* Base input template
=============================== */

import React from 'react';
import { View, StyleSheet, ViewStyle } from 'react-native';
import { Colors } from 'libs/ui/src/WebButton/node_modules/libs/ui/src/WebLoading/node_modules/colibri/styles';
import { AppIcon } from '../../AppIcon';

export interface BaseInputProps {
  style?: ViewStyle;
  icon?: string;
  iconPosition?: 'left' | 'right';
}

export const BaseInput: React.FC<BaseInputProps> = ({
  style,
  children,
  icon,
  iconPosition = 'left',
}) => {
  // styles

  const styles = StyleSheet.create({
    baseInput: {
      backgroundColor: '#fff',
      padding: 15,
      marginTop: 10,
      marginBottom: 10,
      width: '100%',
      borderRadius: 10,
      shadowColor: '#000',
      shadowOffset: { width: 0, height: 4 },
      shadowRadius: 4,
      shadowOpacity: 0.1,
      elevation: 5,
    },
    icon: {
      marginRight: iconPosition === 'left' ? 16 : 2,
      marginLeft: iconPosition === 'left' ? 2 : 16,
    },
  });

  return (
    <View style={[styles.baseInput, style]}>
      {icon && (
        <AppIcon
          name={icon}
          color={Colors.blueLight}
          size={24}
          style={styles.icon}
        />
      )}
      {children}
    </View>
  );
};
