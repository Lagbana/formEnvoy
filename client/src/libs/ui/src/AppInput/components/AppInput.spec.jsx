import React from 'react'
import { shallow, configure } from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16'
import { BaseInput } from './BaseInput'

configure({ adapter: new Adapter() })

test('<BaseInput /> renders correctly', () => {
  const component = shallow(<BaseInput />)
  expect(toJson(component)).toMatchSnapshot()
})
