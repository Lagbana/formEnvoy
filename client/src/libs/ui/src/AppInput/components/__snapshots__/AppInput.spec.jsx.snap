// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BaseInput /> renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "borderRadius": 10,
        "elevation": 5,
        "marginBottom": 10,
        "marginTop": 10,
        "padding": 15,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 4,
          "width": 0,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 4,
        "width": "100%",
      },
      undefined,
    ]
  }
/>
`;
